### DISCUSSION TITLE
Bind dictionary value by key

### DISCUSSION THREAD

**Discussion by Ghostbird** [dotnet/maui#12229] [Q&A]
I've got a dictionary representing a semi-dynamic object. I want to bind to the _Icon_ property.
Aside from the fact that this could be modelled much more elegantly, I'm sure it's possible in XAML to bind to a dictionary key, but I don't know the syntax.

The data has this structure:
```json
{
  "Properties": {
    "Icon": "person-outline"
  }
}
```
Here the root binding context is a proper class, but `Properties` is an `IDictionary<string, object>`.

I tried binding to `Properties[Icon]` but that didn't work.
Does anyone know how this should be done?

**Comment by masterofdaemon** [comment]
it works for me like this:
`  public Dictionary<string, Color> ObservableParameters
        {
            get => Parameters;
            set
            {
                SetProperty(Parameters, value, this, (u, n) => u.Parameters = n);
            }
        }`


but my keys is DateTime, can anyone help with binding?
`                        BackgroundColor="{Binding BindingContext.ObservableParameters[DateTime.Day-DateTime.Month], Source={x:Reference This}}"
`   it works only with key like [9-10]

**Reply by Ghostbird** [reply]
I don't see how that would work. It seems like a convoluted way of what I already have. It would more properly be written as:

```csharp
private IDictionary<string, Color> parameters = new();
public IDictionary<string, Color> ObservableParameters {
  get => parameters;
  set => SetProperty(ref parameters, value);
}
```

Code technically I don't see how using the four parameter overload that passes a model and callback would help, except it makes it so that you don't have to use the `ref` keyword. AFAIK the only reason why you'd do that is if `Parameters` is a C# property, because then you can't use `ref`. However if you that, you've got bigger problems, since your binding won't update if you write directly to `Parameters`. This property pattern should always be backed by a private member that is never accessed outside of this pattern. Hence I used that approach.

Functionally this makes it possible to bind to the dictionary itself, but won't help when trying to bind to a dictionary entry, since there's no code that interacts with those.

**Reply by masterofdaemon** [reply]
Yes, i need to make work days the different color:

`  WorkDays.Add(new DateTime(2023, 2, 11));

            WorkDays.Add(new DateTime(2023, 2, 12));

            foreach (var day in Calendar.Days)

            {

                if (WorkDays.Contains(day.DateTime))

                {

                    Parameters.Add($"{day.DateTime.Day}-{day.DateTime.Month}-{day.DateTime.Year}", Color.FromArgb("#00f"));

                }
                else

                {
                    Parameters.Add($"{day.DateTime.Day}-{day.DateTime.Month}-{day.DateTime.Year}", Color.FromArgb("#f00"));

                }
            }`
but i can't bind that dates in xaml
